opsdroid/connector/matrix/connector.py

```diff
@@ -112,11 +154,31 @@ def filter_json(self):
             if display_name != self.nick:
                 await self.connection.set_display_name(self.mxid, self.nick)
 
+    async def disconnect(self):
+        """Close the matrix session."""
+        await self.session.close()
+
     async def _parse_sync_response(self, response):
         self.connection.sync_token = response["next_batch"]
-        for roomid in self.room_ids.values():
-            room = response["rooms"]["join"].get(roomid, None)
-            if room and "timeline" in room:
+
+        # Emit Invite events for every room in the invite list.
+        for roomid, room in response["rooms"]["invite"].items():
+            # Process the invite list to extract the person who invited us.
+            invite_event = [
+                e
+                for e in room["invite_state"]["events"]
+                if "invite" == e.get("content", {}).get("membership")
+            ][0]
```

Comment:
This is quite the list comprehension. Is there not a way to do this more concisely?

Reply:
I can't think of one, it's parsing quite a complex response structure in the /sync feed.